### PR TITLE
refactor!: Namegenerator functions refactor

### LINKED ISSUE #9: Refactor NameGenerator functions
### Is your feature request related to a problem?

Current approach have 2 separate function for generating names for AWS and Azure, this are very similar and can be merge into one function. 

### Describe the solution you'd like

Code refactor.

### Describe alternatives you've considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by horiagunica**
## Description

Refactor Name Generator function into a single function. Also introduce obtaining GCP project ID based on repo secrets environment variables.

This PR closes #9  .

## Motivation and Context

Code refactoring.

## How Has This Been Tested?

Tested using separate test repo : https://github.com/PaloAltoNetworks/test-gcp-ci-workflows/ .

Tested by pointing the GO module to the specific branch .

## Types of changes

- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pkg/testskeleton/testskeleton.go [hunk 1]
```diff
@@ -58,47 +58,25 @@ type AdditionalChangesAfterDeployment struct {
...
-}
-
-// Function that generates and returns a set of random Azure resource names.
-// Randomization is based on UUID.
-func GenerateAzureRandomNames() AzureRandomNames {
-	prid := os.Getenv("PRID")
-	if prid != "" {
-		prid = fmt.Sprintf("-pr%s-", prid)
-	}
+	project_id := os.Getenv("PROJECT_ID")
```
Comment (sebastianczech) [inline pkg/testskeleton/testskeleton.go]: Can we use [camel case for naming variables](https://betterprogramming.pub/naming-conventions-in-go-short-but-descriptive-1fa7c6d2f32a) ? e.g.  

```suggestion
	projectId := os.Getenv("PROJECT_ID")
```

File: pkg/testskeleton/testskeleton.go [hunk 2]
```diff
@@ -107,10 +85,11 @@ func GenerateAzureRandomNames() AzureRandomNames {
...
 
-	names := AzureRandomNames{
-		NamePrefix:         fmt.Sprintf("ghci%s%s-", prid, prefixId),
-		ResourceGroupName:  strings.Join(gid, ""),
-		StorageAccountName: fmt.Sprintf("ghci%s", strings.Join(storageId, "")),
+	names := TerraformVarsInfo{
+		NamePrefix:              fmt.Sprintf("%s-%s-", prid, prefixId),
+		AzureResourceGroupName:  strings.Join(gid, ""),
+		AzureStorageAccountName: fmt.Sprintf("ghci%s", strings.Join(storageId, "")),
+		GCPProjectID:            project_id,
```
Comment (sebastianczech) [inline pkg/testskeleton/testskeleton.go]: If in line 79 we are going to use camel case, then here we also need to update it into:

```suggestion
		GoogleProjectId:            projectId,
```

Additionally I replaced `GCP` by `Google`.

Do you accept that approach ?

**Review by sebastianczech — COMMENTED** [review]
Great job, thank you for refactoring code and doing 1 function (instead multiple ones per cloud) 👍 

Besides below comments, I have also 1 more question - what is the reason to pass `PROJECT_ID` via Terratest skeleton , do not change anything in that value and return it in struct `TerraformVarsInfo` ? Maybe we could directly get project ID in tests in repository `terraform-google-vmseries-modules`, without passing that value into skeleton and reading it from outputs ?

**Review by sebastianczech — APPROVED** [review]
great job 👍 
please change PR title from `refactor:` into `refactor!:` as we have breaking change.

**Review by sebastianczech — APPROVED** [review]
👍

**Review by FoSix — APPROVED** [review]
👍🏻

**Pull request merged**